core/src/test/java/hudson/util/io/ZipArchiverTest.java

```diff
@@ -81,26 +79,25 @@ public void huge64bitFile() throws IOException {
         }
     }
 
-    @Ignore("TODO fails to add empty directories to archive")
     @Issue("JENKINS-49296")
     @Test
     public void emptyDirectory() throws Exception {
         Path zip = tmp.newFile("test.zip").toPath();
         Path root = tmp.newFolder().toPath();
-        Files.createDirectory(root.resolve("foo"));
-        Files.createDirectory(root.resolve("bar"));
-        Files.writeString(root.resolve("bar/file.txt"), "foobar", StandardCharsets.UTF_8);
+        Files.createDirectory(root.resolve("a"));
+        Files.createDirectory(root.resolve("b"));
+        Files.writeString(root.resolve("a/file.txt"), "empty_dir_zip_test");
         try (OutputStream out = Files.newOutputStream(zip)) {
             new FilePath(root.toFile()).zip(out, "**");
         }
-        Set<String> names = new HashSet<>();
-        try (InputStream is = Files.newInputStream(zip);
-             ZipInputStream zis = new ZipInputStream(is, StandardCharsets.UTF_8)) {
-            ZipEntry ze;
-            while ((ze = zis.getNextEntry()) != null) {
-                names.add(ze.getName());
+        Set<String> actual = new HashSet<>();
+        Set<String> expected = Set.of("a/", "b/", "a/file.txt");
+        try (ZipInputStream zipInputStream = new ZipInputStream(Files.newInputStream(zip), StandardCharsets.UTF_8)) {
```

Comment:
does the wrapped input stream also get closed? i.e. why the change in behaviour from before which closed both input streams?

Reply:
it was mainly that I manually added a test which then appeared in a similar way when I merged master into my branch. But you are right, it doesn't close. I will change that